src/envoy/auth/jwt_test.cc

```diff
@@ -307,242 +316,206 @@ class JwtTestWithJwk : public testing::Test {
+               nullptr);
+    }
+    {
+      // JsonBadInputHeader
+      add_test(ds.kJwtWithBadJsonHeader, ds.kPublicKey, "pem", false,
+               Status::JWT_HEADER_PARSE_ERROR, nullptr);
+    }
+    {
+      // JsonBadInputPayload
+      add_test(ds.kJwtWithBadJsonPayload, ds.kPublicKey, "pem", false,
+               Status::JWT_PAYLOAD_PARSE_ERROR, nullptr);
+    }
+    {
+      // AlgAbsentInHeader
+      add_test(ds.kJwtWithAlgAbsent, ds.kPublicKey, "pem", false,
+               Status::JWT_HEADER_NO_ALG, nullptr);
+    }
+    {
+      // AlgIsNotString
+      add_test(ds.kJwtWithAlgIsNotString, ds.kPublicKey, "pem", false,
+               Status::JWT_HEADER_BAD_ALG, nullptr);
+    }
+    {
+      // InvalidAlg
+      add_test(ds.kJwtWithInvalidAlg, ds.kPublicKey, "pem", false,
+               Status::ALG_NOT_IMPLEMENTED, nullptr);
+    }
+  }
+  {
+    DatasetJwk ds;
+    {
+      // JwtDecodeWithJwk
+      auto payload = Json::Factory::loadFromString(ds.kJwtPayload);
+      add_test(ds.kJwtNoKid, ds.kPublicKey, "jwks", true, Status::OK, payload);
+    }
+    {
+      // CorrectKid
+      auto payload = Json::Factory::loadFromString(ds.kJwtPayload);
+      add_test(ds.kJwtWithCorrectKid, ds.kPublicKey, "jwks", true, Status::OK,
+               payload);
+    }
+    {
+      // IncorrectKid
+      add_test(ds.kJwtWithIncorrectKid, ds.kPublicKey, "jwks", false,
+               Status::JWT_INVALID_SIGNATURE, nullptr);
+    }
+    {
+      // NonExistKid
+      add_test(ds.kJwtWithNonExistKid, ds.kPublicKey, "jwks", false,
+               Status::KID_ALG_UNMATCH, nullptr);
+    }
+    {
+      // BadFormatKid
+      add_test(ds.kJwtWithBadFormatKid, ds.kPublicKey, "jwks", false,
+               Status::JWT_HEADER_BAD_KID, nullptr);
+    }
+    {
+      // JwkBadJson
+      std::string invalid_pubkey = "foobar";
+      add_test(ds.kJwtNoKid, invalid_pubkey, "jwks", false,
+               Status::JWK_PARSE_ERROR, nullptr);
+    }
+    {
+      // JwkNoKeys
+      std::string invalid_pubkey = R"EOF({"foo":"bar"})EOF";
+      add_test(ds.kJwtNoKid, invalid_pubkey, "jwks", false, Status::JWK_NO_KEYS,
+               nullptr);
+    }
+    {
+      // JwkBadKeys
+      std::string invalid_pubkey = R"EOF({"keys":"foobar"})EOF";
+      add_test(ds.kJwtNoKid, invalid_pubkey, "jwks", false,
+               Status::JWK_BAD_KEYS, nullptr);
+    }
+    {
+      // JwkBadPublicKey
+      std::string invalid_pubkey = R"EOF({"keys":[]})EOF";
+      add_test(ds.kJwtNoKid, invalid_pubkey, "jwks", false,
+               Status::JWK_NO_VALID_PUBKEY, nullptr);
+    }
+  }
+
+  return testdata;
 }
 
-TEST_F(JwtTestWithJwk, JwkBadPublicKey) {
-  std::string invalid_pubkey = R"EOF({"keys":[]})EOF";
-  JwtVerifier v;
-  auto payload = v.Decode(*Pubkeys::ParseFromJwks(invalid_pubkey), kJwtNoKid);
-  EXPECT_FALSE(payload);
-  EXPECT_EQ(v.GetStatus(), Status::JWK_NO_VALID_PUBKEY);
-}
+INSTANTIATE_TEST_CASE_P(TestCases, JwtTest, testing::ValuesIn(TestData()));
```

Comment:
I am worrying about if one of tests fail,  how easy to identify which one file,  does each test has a name to associate with it?

Reply:
No, each test has only a number (corresponding to what-th test it is), and I agree that it is not so easy to see if it fails...

Reply:
IMO it is fine with numbers.

Reply:
I fixed it anyway